packages/running-extension/src/kernels.tsx

```diff
@@ -41,6 +44,9 @@ export async function addKernelRunningSessionManager(
   const { runningChanged, RunningKernel } = Private;
   const throttler = new Throttler(() => runningChanged.emit(undefined), 100);
   const trans = translator.load('jupyterlab');
+  const shutdownUnusedLabel = trans.__('Shut Down Unused');
+  const runningKernels = await KernelAPI.listRunning();
```

Comment:
when I moved everything to this file, runningKernels is not refresh the same as the other file
this file seems to be on page reload, while the other seems to be on kernel update

also the kernels section got moved to the end of the list for some reason

Reply:
`CommandToolbarButton` may be easier to use use here, although it will require defining a command. It would be easier because the commands have `isVisible() => boolean` and `isEnabled()` functions that the `CommandToolbarButton` will respect. If you need more hints on this one let me know.

> also the kernels section got moved to the end of the list for some reason

maybe do to the await in this line (`await KernelAPI.listRunning();`)?